### ISSUE TITLE
Is the project abandoned?

### ISSUE THREAD

**Issue by jaklan** [noirbizarre/flask-restplus#593]
The last commit was made on Oct 1, 2018 - almost 5 moths ago. Does it mean the project is actually dead?

**Comment by tincumagic** [comment]
This is not quite a response you expect but ...
As you say creator noirbizarreo no longer does any commit for us but I'm actively using this self documenting framework and so are others and I would like to contribute, and the project could continue to evolve with individual contributions of it's users ... so there is hope.

**Comment by hoenie-ams** [comment]
Yes @tincumagic , fully agree!

**Comment by hygorxaraujo** [comment]
@tincumagic, I'm also still actively using Flask-RESTPlus and would like to contribute and see it keep growing, but without @noirbizarre is there anyone else who can accept PRs and make new releases?

**Comment by jaklan** [comment]
You can always make a fork...

**Comment by SteadBytes** [comment]
@tincumagic I agree too! I'm actively using Flask-RESTPlus and would love to help grow and evolve the project. @noirbizarre would you be willing to allow others from the community to have write access?

**Comment by dwd-umd** [comment]
Thank you to those who are offering to step up to the plate. It would be reassuring to see some resolution on the future of this project. I used flask_restplus in my last project but, as things stand, I'll probably not use it again if it's not being maintained. Is anyone here actually in contact with @noirbizarre ?

**Comment by SteadBytes** [comment]
@dwd-umd I'm in the same boat, I really like this framework and have been using it on several projects. However, if we are unable to maintain the project I will likely not use it again which would be sad to have to do. I'm not in touch with @noirbizarre unfortunately, though I think he is still active in other projects on GitHub.

**Comment by ziirish** [comment]
Other and I already had a similar discussion several months ago on the project gitter.

The best solution would indeed to have several people able to commit and the project and being able to update the pypi packages.
The alternative being to fork the project I guess.

@noirbizarre any thoughts?

BTW thanks for all the work you already put into this nice and useful project!

**Comment by SteadBytes** [comment]
Yes @ziirish thats definitely the best solution 👍 It would be a shame to have to fork the project and build up a community again 🙁

**Comment by ziirish** [comment]
Any volunteer to keep maintaining the project?

I'd say we need at least 2-3 active contributors.

There is probably a lot of bug triaging to begin with + some PR to review.

I personally have very few spare time, but since I'm using Flask-RESTplus both for personal and professional projects I may give a hand.

**Comment by SteadBytes** [comment]
@ziirish I would be more than happy to volunteer 👍

**Comment by hoenie-ams** [comment]
Me too!

**Comment by Ryanglambert** [comment]
Same here.

This project has a lot of potential.

Perhaps Noirbizzare just got burnt out

**Comment by ziirish** [comment]
Ok, here is how I see things. Let me know what you think about it.

1. First we need to get in touch with @noirbizarre We need to know whether:
  - he wants to keep maintaining the project by his own
  - he wants to onboard new maintainers
  - he wants to stop maintaining the project and give it to the community
  - he doesn't want to do anything
2. Depending on what is decided above, we may want to choose a few maintainers: maybe we can open an issue where people can volunteer with a small presentation post, then the community vote with the :+1: emoji

@SteadBytes you seem quite involved lately, do you want to take the lead on this?

**Comment by ramarivera** [comment]
I would also like to help mantain it! (y)

**Comment by YJinHai** [comment]
Looking forward to the results

**Comment by SteadBytes** [comment]
@ziirish Yes I'd be more than happy to do that. The only issue is actually getting in touch with @noirbizarre 🤔 He has no email advertised on his account or website and his Twitter DMs aren't open. I might have to try and publicly tweet at him unfortunately. I will see what I can do and keep this thread updated 👍

**Comment by SteadBytes** [comment]
Just to keep everyone in the loop, I tweeted @noirbizarre on Saturday morning. I haven't received a reply yet. https://twitter.com/bensteadman_/status/1106960333111992321?s=19

**Comment by nvgoldin** [comment]
Hey guys, in the same boat here. I'd be happy to help out - I won't have much time to "triage/maintain" but can work on any specific small-scoped task as needed (from bugs, features, CI and deployment) - assuming we reach to a flow where PR can get in.

**Comment by YJinHai** [comment]
Let me see if I can make any progress

**Comment by SteadBytes** [comment]
@YJinHai Are you in contact with @noirbizarre ?

**Comment by noirbizarre** [comment]
Hi everyone :wave: 

Sorry for not being present lately.

Just so you know, I want to keep maintaining the project but I want to onboard some maintainers too.

There wasn't a lot of public activity lately because:
- I didn't had as much time as I want to work on `flask-restplus`
- I was trying to solve some performance problems requiring refactoring which is difficult to do wile accepting pull requests.

So, I was thinking:
- We need to choose some maintainers (I don't know how, but @ziirish  and @SteadBytes seems to have been very active lately)
- We need to decide how we want to organize the cooperation

At least, as may problem is time availability, I was thinking about some sponsoring (being paid to work on flask-restplus as today the main problem is that I can't priorize flask-restplus above my clients). I don't know yet how to do that, but I'm thinking about (maybe opening a patreon).

I'm open to any contribution, any remark or other proposal to improve the current state (don't hesitate to use [![Gitter](https://badges.gitter.im/noirbizarre/flask-restplus.svg)](https://gitter.im/noirbizarre/flask-restplus), I will answer)

**Comment by SteadBytes** [comment]
Hey @noirbizarre, it's great to hear from you and that you want to continue maintaining the project - need to apologise! 😊

Onboarding new maintainers and organising cooperation is tricky indeed  🤔 Do you have any criteria for maintainers and how many? I would very much like to be involved if you're willing to have me and I can meet those criteria. Im sure there are others too.

I think with a small set of maintainers a roadmap/feature priority can be established. These could then be made into issues and implemented individually and PRs reviewed. PRs not directly tied to a specific road map issue (i.e. from the community) could then be reviewed and accepted if some minimum number of maintainers agree on it?

Sponsoring would of course be amazing if possible, would that go towards paying your time working on the project?

And finally, I'd definitely be interested to know what the performance improvements you're working on are 😁

**Comment by ziirish** [comment]
Hi @noirbizarre 

I have a very few spare time (that I already spend to contribute to some other projects) but I am willing to give you a hand. I thought I could start by doing some bug triaging, then review some PRs.

I don't know what workflows GitHub allows you to setup but as suggested by @SteadBytes setting up a minimal number of reviewers to accept a PR is a good start I guess.

**Comment by j5awry** [comment]
Hi, I've been a bit of a quiet lurker here. My company is using flaskRestPlus for new services, and I could go a similar route of asking for some time to be a maintainer. Not sure if I'd be able to get that as part of the assignment, but I'm willing to try.

**Comment by a-luna** [comment]
Hello, everyone! I, too, use and greatly appreciate the work that has been put into flask-restplus and would be interested in helping out in any way. I am new to the world of OSS and have (embarrassingly) never collaborated on a project. I would love to learn the ropes by doing any type of maintenance work, no matter how tedious or menial.

Please let me know how I can be of service to keep this awesome extension alive and thriving!

**Comment by noirbizarre** [comment]
Hey everybody !

The opening time has come and I added some maintainers:
- @ziirish 
- @SteadBytes 
- @j5awry 
- @a-luna 

As a start, bug triage seems the way to go:
- it will allows you/us to discover the issues, discuss them...
- it may be the base of a roadmap (even if I already something in my head)

Don't hesitate to adds some missing labels.

Then we can see which PR can be merged, which one only needs documentation and which one only needs closing.

Maybe we can try to have a first talk all together, either by raw chat or webrtc next week so we can discuss about those issue and how we want to organize ourself. I can share my ideas for a roadmap...

Anyway, I'm really open to discussion and suggestion.

Just so I know, where are you from ? On which timezone ? (to be able to find a moment to talk)

Me I'm french, from Paris suburbs (which explain the english typos :trollface: ) and so I'm on GMT+1.

**Comment by SteadBytes** [comment]
Thank you very much for adding me as a maintainer @noirbizarre , I look forward to working together with you and the others on the project.

Bug triaging sounds like a great way to start, definitely needed to put together a roadmap. You mentioned you may have an outline in your head, are you willing to share that as it may have some bearing on prioritising bugs/PRs?

I am from Colchester in England (near London) so I'm GMT timezone. More than happy to find time to talk! 😊

**Comment by ziirish** [comment]
Thanks @noirbizarre 
I'll start to review and triage the opened issues ASAP.

Despite my nickname, I'm French from Northern France.

**Comment by j5awry** [comment]
Hey all. I'm based in Kansas City, Missouri, so -6 GMT, working for a company out of Boston, MA (listed in my profile). I'm still waiting on the official "legal" things to come through on my end (saying it's ok to do work, not get in trouble with my non-compete, etc.) but i'm not too worried about it. My boss is behind it. 

Point me in a direction and I can get going. Our teams pain is definitely on the marshalling and api doc end -- making sure everything is happy with Marshamallow, jsonschema, etc. But that's just _our_ priorities. If more people on my team want it, they're welcome to join in the fun.

Also, my French is rusty, but I'm willing to try :)

**Comment by a-luna** [comment]
Hi everyone, I live in Reno, Nevada (-7 GMT currently, -8 GMT when daylight savings ends). Thank you for adding me as a contributor, @noirbizarre, I am available anytime (day or night) to meet and discuss the best way I can contribute!

**Comment by SteadBytes** [comment]
@j5awry My French is also somewhat rusty! 😅 Might have to download Duolingo hehe. In terms of priorities on what to work on I think we're first going to sort out the bugs/PRs currently open and then build a roadmap to move forwards from there. My company also have some areas of priority but we need to decide as a community on where to go next and what to prioritise 👍 Maybe it's worth listing in a comment on #612 the features/issues that we each think are worth considering for a roadmap?

**Comment by NiharikaSinghal** [comment]
@noirbizarre , i started using this project few weeks ago, i don't have much knowledge, but i would be happy to help, if you need any help from my side. I can spare sometime in a week

**Comment by mutoromike** [comment]
Hello @noirbizarre, hello everyone. I've also been closely following up on the conversation and would be more than glad to help in whichever way I can. I've been using Flask-RestPlus for a while and would love to see it turning into the most awesome version of Flask(it already is). I am very much available.

**Comment by lrabbt** [comment]
Hey, everyone! I've been using this project for a few weeks and i have to say I love the idea behind it and appreciate all the effort put into it. @noirbizarre, I never really contributed to an open-source project, but I'd love to try and help with the growing of this one and can definitely spare some time on my schedule for it :+1:

**Comment by michaelbukachi** [comment]
@noirbizarre My company also uses Flask-Restplus a lot. I'd like to volunteer to become a maintainer also. I'll start making pull requests

**Comment by jhallard** [comment]
I'd like to chime in that the company I work for also has production services running on top of Flask-Restplus and that we enjoy this project quite a bit. I'd also like to start making some pull requests in the future.

**Comment by andreixk** [comment]
I create a pull request almost a month ago and it's been pretty much ignored (there are currently 60 open PRs). Even if this project is not dead, I would strongly advise against using it in production.

**Comment by michaelbukachi** [comment]
The current project maintainers should give us feedback of what's going on. @ziirish @SteadBytes @j5awry @a-luna do you mind chipping in?

**Comment by SteadBytes** [comment]
@andreixk I appreciate your frustration and apologise for not reviewing your PR. Please understand that there is a backlog of issues/pull requests which we are attempting to work through whilst struggling greatly to communicate with @noirbizarre in order to make releases of the project as only @noirbizarre has access to push new releases to PyPi - we are currently hesitant to continue merging changes due to this https://github.com/noirbizarre/flask-restplus/issues/743 .

> I would strongly advise against using it in production
You are more than welcome to your opinions about Flask RESTPlus, however if you feel the need to make statements such as this can you please at the very least explain why you feel this is case? Otherwise, please don't :+1: 

@michaelbukachi Most discussion has been taking place over in [Gitter](https://gitter.im/noirbizarre/flask-restplus) and on the issue created by @j5awry to make the next release https://github.com/noirbizarre/flask-restplus/issues/743 . I have left a [comment](https://github.com/noirbizarre/flask-restplus/issues/743#issuecomment-557906090) on that issue outlining the current state of affairs with a summary of the ongoing discussion on Gitter to make it more visible as I appreciate that it's difficult to track things via Gitter :+1:

**Comment by andreixk** [comment]
> You are more than welcome to your opinions about Flask RESTPlus, however if you feel the need to make statements such as this can you please at the very least explain why you feel this is case? Otherwise, please don't 👍

In the previous statement I stated that there are 60 open PRs. If a project has PRs dating back almost 4 years, it's a sign that a project has few contributors. I'm not bashing it or saying you need to do a better job. I understand that this is an open-source collaboration project, and everyone is busy with their day jobs, but they're doing their best. 
I'm simply stating a fact: there is not enough resources on this project to allow it to support a critical production-level environment.

**Comment by SteadBytes** [comment]
@andreixk Thank you for the explanation, I appreciate it. None of us want the project to be unmaintained, please see the [comment](https://github.com/noirbizarre/flask-restplus/issues/743#issuecomment-557906090) I linked previously for the explanation of the current situation. If @noirbizarre returns and provides us with the capability to progress (mainly making releases) we will be able to get back on track 👍

**Comment by michaelbukachi** [comment]
@SteadBytes Thanks. Following.

**Comment by rakeshsehgal1** [comment]
Hello,

I wanted to know if anyone has used this in production? I started using this but found the documentation is very poor, examples very basic, no project template at all and looks like from the "pin" it was abandoned at some point.

A very basic use case of nested fields in model has to be Googled which takes you to a stackoverflow [link](https://stackoverflow.com/questions/54263231/model-definition-of-array-data-in-python-flask-restplus-swagger) which takes you to a git issue which again takes you to a stackoverflow link for solution. It's probably OK to use to learn how to design API, etc.

I started wondering that even though there is a roadmap, does anyone recommend using this in production (to the point @andreixk is trying to make)? Why wouldn't one use swagger editor to download code in 20+ languages (with docker file and test code!), cwiki documentation, mysql schema, protobuf and a lot more? After all, isn't writing less code better?

Please this is not to offend anyone especially the authors (great effort!), just trying to gather some thoughts.

**Comment by jhallard** [comment]
@rakeshsehgal1 I am using this project in production for some tools that are used internally at the company that I current work for. It hasn't been without hiccup (mostly integrating with Google oAuth flow and the `flask-dance` library), but otherwise, I've been able to build a complex, fully-features, and highly-reliable API with this project. 

If I were to start again from scratch, would I use this project? Possibly not... given the state of it and the slow code velocity, but I have been able to make it work and work well.

**Comment by Colin-b** [comment]
@rakeshsehgal1 we use it in production, it's stable enough, but indeed, as stated by numerous people, the slow velocity of this project (no blame here, we all know what we do when we use free projects, people have lives outside of code ;-) ) and the fact that ASGI is what we look for nowadays makes that I would not recommend it for new projects from scratch (this recommendation stands for every flask based fmk really).

**Comment by michaelbukachi** [comment]
@rakeshsehgal1 We also use it in production for a number of internal APIs. Works really well. Though we had to implement quite a number of workarounds. We might shift to something else in the future. Still doing research on possible alternatives.

**Comment by mervynrussell** [comment]
Been using it in production systems for approx 4 years.  Works great, no major issues that couldn't be worked around.

**Comment by CrazyBonze** [comment]
I personally would like to see the velocity for this project increase and would even contribute back to it because i have used this for an internal project at work for the last 1.5 years as well as use it for my own personal projects.

**Comment by rns4731** [comment]
+1 to keeping this project alive, willing to help if I can.
